spec/services/cleaners/interim_claim_cleaner_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+require 'services/cleaners/cleaner_shared_examples'
+
+RSpec.describe Cleaners::InterimClaimCleaner do
+  subject(:cleaner) { described_class.new(claim) }
+
+  describe '#call' do
+    subject(:call_cleaner) { cleaner.call }
+
+    let(:claim) { create(:interim_claim, disbursements: build_list(:disbursement, 1)) }
+
+    before { claim.fees << build(:interim_fee, fee_type, claim: claim) }
+
+    context 'without a warrant fee' do
+      let(:fee_type) { :disbursement }
+
+      it { expect { call_cleaner }.not_to change { claim.disbursements.size }.from(1) }
+    end
+
+    context 'with a warrant fee' do
+      let(:fee_type) { :warrant }
+
+      it { expect { call_cleaner }.to change { claim.disbursements.size }.from(1).to(0) }
```

Comment:
yes, this looks correct logic to me. interim claims for warrants cannot claim disbursements, but interim claims for  Trial start, Effective PCMH or "Disbursement only" can claim them. 

If you choose one of the last three and add a disbursement then navigate back and change to a warrant then the disbursements need to be cleaned/deleted.